docs/architecture/adr-005-subscription.md

```diff
@@ -0,0 +1,225 @@
+# ADR 5: Subscription
+
+## Changelog
+
+
+## Context
+
+Cosmos-SDK should allow developers to choose from a variety of payment models for their application. One of the most popular payment models for Internet applications, so it is natural to provide this payment model as a module for sdk users.
+
+For fuller context around this issue: see [\#4642](https://github.com/cosmos/cosmos-sdk/issues/4642)
+
+## Decision
+
+Create Subscription module in SDK so that users can subscribe to various subscription services. Subscription collectors will send messages to collect subscription payments from users who have due subscriptions. Subscriptions are paid out directly from a user account. If the user does not have enough funds to pay for the subscription is inactivated. Users can define maximum limits on how many periods the subscription is valid for; if this limit elapses before a subscription renewal, the subscription is invalidated.
+
+## Status
+
+#### Proposed
+
+## Implementation Changes
+
+Introduce the following messages to `x/subscription` module:
+
+```go
+// CreateSubscriptionMsg allows Service Provider to create new subscription service.
+type CreateSubscriptionMsg struct {
+    Name      string         // unique, human-readable name for subscription service
```

Comment:
I think this shouldn't be unique and we should have an independent unique `string ID`. The `Name` would be a separate property that's not necessarily unique